internal/config/config.go

```diff
@@ -40,4 +40,6 @@ type Config struct {
 	ResyncPeriod time.Duration
 	// MaximumRetry defined how many times we retry if reconciliation failed.
 	MaximumRetry int
+	// Timeout defines the general timeout for the reconciliation.
+	Timeout time.Duration
```

Comment:
Can we add a more descriptive name for that? Reading in the `internal/config` package "Timeout" feels like "timeout for what?"